### PR TITLE
feat: create controller for result entity implementing all crud function

### PR DESCRIPTION
**Pull request by blertaazemi**
…ions

### CODE REVIEW

File: src/OnlineVotingS.API/Controllers/ResultController.cs [hunk 1]
```diff
@@ -0,0 +1,283 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using OnlineVotingS.Application.DTO.PostDTO;
+using OnlineVotingS.Application.DTO.PutDTO;
+using OnlineVotingS.Application.Services.Results.Requests.Commands;
+using OnlineVotingS.Application.Services.Results.Requests.Queries;
+using OnlineVotingS.Domain.Entities;
+
+namespace OnlineVotingS.API.Controllers
+{
```
Comment (AdhurimUP) [inline src/OnlineVotingS.API/Controllers/ResultController.cs]: remove brackets

File: src/OnlineVotingS.API/Controllers/ResultController.cs [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    [ApiController]
+    [Route("api/[controller]")]
+    public class ResultController : ControllerBase
+    {
```
Comment (AdhurimUP) [inline src/OnlineVotingS.API/Controllers/ResultController.cs]: align code please

File: src/OnlineVotingS.API/Controllers/ResultController.cs [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+                    result.Candidates.FullName,
+                    result.Candidates.Party,
+                    result.Candidates.Description,
+                    result.Candidates.Income,
+                    result.Candidates.Works
+                } : null
+            });
+
+            return Ok(response);
+        }
```
Comment (AdhurimUP) [inline src/OnlineVotingS.API/Controllers/ResultController.cs]: new empty line at bottom

File: src/OnlineVotingS.API/Controllers/ResultController.cs [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+        [HttpGet("{resultId:int}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<object>> GetResultById([FromRoute] int resultId)
```
Comment (AdhurimUP) [inline src/OnlineVotingS.API/Controllers/ResultController.cs]: ideally we name int id, but as u prefer you can leave it

File: src/OnlineVotingS.API/Controllers/ResultController.cs [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+        {
+            var query = new GetResultByIdQuery(resultId);
+            var result = await _mediator.Send(query);
+
+            if (result == null)
+            {
+                return NotFound($"Result with ID {resultId} not found.");
+            }
+
+            // Ensuring related entities are not null before accessing them
```
Comment (AdhurimUP) [inline src/OnlineVotingS.API/Controllers/ResultController.cs]: remove this

File: src/OnlineVotingS.API/Controllers/ResultController.cs [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+                    result.Candidates.Description,
+                    result.Candidates.Income,
+                    result.Candidates.Works
+                } : null
+            };
+
+            return Ok(response);
+        }
+
+
```
Comment (AdhurimUP) [inline src/OnlineVotingS.API/Controllers/ResultController.cs]: remove a empty line

File: src/OnlineVotingS.API/Controllers/ResultController.cs [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+                    result.Candidates.Description,
+                    result.Candidates.Income,
+                    result.Candidates.Works
+                } : null
+            });
+
+            return Ok(response);
+        }
+
+
```
Comment (AdhurimUP) [inline src/OnlineVotingS.API/Controllers/ResultController.cs:31]: remove an empty line

File: src/OnlineVotingS.API/Controllers/ResultController.cs [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
```
Comment (AdhurimUP) [inline src/OnlineVotingS.API/Controllers/ResultController.cs:51]: remove an empty line

File: src/OnlineVotingS.API/Controllers/ResultController.cs [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+                    result.Candidates.Description,
+                    result.Candidates.Income,
+                    result.Candidates.Works
+                } : null
+            });
+
+            return Ok(response);
+        }
+    
+
```
Comment (AdhurimUP) [inline src/OnlineVotingS.API/Controllers/ResultController.cs:59]: remove an empty line

File: src/OnlineVotingS.Application/DTO/PutDTO/ResultPutDTO.cs [hunk 10]
```diff
@@ -12,4 +12,5 @@ public class ResultPutDTO
         public int ElectionID { get; set; }
         public int CandidateID { get; set; }
         public int TotalVotes { get; set; }
+
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/DTO/PutDTO/ResultPutDTO.cs:15]: remove an empty line

File: src/OnlineVotingS.Application/Services/Results/Handlers/Queries/GetResultByIdHandler.cs [hunk 11]
```diff
@@ -3,36 +3,40 @@
...
 using OnlineVotingS.Domain.Interfaces;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
 
-namespace OnlineVotingS.Application.Services.Results.Handlers.Queries;
-
-public class GetResultByIdHandler : IRequestHandler<GetResultByIdQuery, Result>
+namespace OnlineVotingS.Application.Services.Results.Handlers.Queries
 {
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/Services/Results/Handlers/Queries/GetResultByIdHandler.cs]: remove a brackets

File: src/OnlineVotingS.Domain/Interfaces/IResultRepository.cs [hunk 12]
```diff
@@ -10,9 +10,11 @@ namespace OnlineVotingS.Domain.Interfaces;
...
 {
+    Task<Result> GetByIdAsync(int id);
     Task<IEnumerable<Result>> GetByElectionIdAsync(int electionId);
-
     Task<IEnumerable<Result>> GetByCandidateIdAsync(int candidateId);
-
     Task<IEnumerable<Result>> GetByTotalVotesGreaterThanAsync(int votes);
+    Task<Result> GetResultWithDetailsAsync(int resultId);
+    Task<IEnumerable<Result>> GetAllResultsWithDetailsAsync(); // Add this line
+
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Domain/Interfaces/IResultRepository.cs]: remove a empty line

File: src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs [hunk 13]
```diff
@@ -6,26 +6,67 @@
 using System.Linq;
 using System.Threading.Tasks;
 
-namespace OnlineVotingS.Infrastructure.Repositories;
-
-public class ResultRepository : GenericRepository<Result>, IResultRepository
+namespace OnlineVotingS.Infrastructure.Repositories
 {
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs]: remove a brackets

File: src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs [hunk 14]
```diff
@@ -6,26 +6,67 @@
...
+             .Include(r => r.Elections)
+             .Include(r => r.Candidates)
+             .ToListAsync();
+        }
 
-    public async Task<IEnumerable<Result>> GetByCandidateIdAsync(int candidateId)
-    {
-        return await _dbSet.Where(r => r.CandidateID == candidateId).ToListAsync();
-    }
+        // Method to get a specific result by ID with related Elections and Candidates entities
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs]: remove a message

File: src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs [hunk 15]
```diff
@@ -6,26 +6,67 @@
...
-    }
+        public ResultRepository(ApplicationDbContext context) : base(context)
+        {
+        }
 
-    public async Task<IEnumerable<Result>> GetByElectionIdAsync(int electionId)
-    {
-        return await _dbSet.Where(r => r.ElectionID == electionId).ToListAsync();
-    }
+        // Method to get all results with related Elections and Candidates entities
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs]: remove a message

File: src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs [hunk 16]
```diff
@@ -6,26 +6,67 @@
...
+            return await _dbSet
+                .Include(r => r.Elections)  // Including related Elections entity
+                .Include(r => r.Candidates) // Including related Candidates entity
+                .FirstOrDefaultAsync(r => r.ResultID == resultId);
+        }
 
-    public async Task<IEnumerable<Result>> GetByTotalVotesGreaterThanAsync(int votes)
-    {
-        return await _dbSet.Where(r => r.TotalVotes > votes).ToListAsync();
+        // Method to get results by Election ID with related entities
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs]: remove message

File: src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs [hunk 17]
```diff
@@ -6,26 +6,67 @@
...
+        public async Task<IEnumerable<Result>> GetByElectionIdAsync(int electionId)
+        {
+            return await _dbSet
+                .Include(r => r.Elections)
+                .Include(r => r.Candidates)
+                .Where(r => r.ElectionID == electionId)
+                .ToListAsync();
+        }
+
+        // Method to get results by Candidate ID with related entities
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs]: remove message

File: src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs [hunk 18]
```diff
@@ -6,26 +6,67 @@
...
+        public async Task<IEnumerable<Result>> GetByCandidateIdAsync(int candidateId)
+        {
+            return await _dbSet
+                .Include(r => r.Elections)
+                .Include(r => r.Candidates)
+                .Where(r => r.CandidateID == candidateId)
+                .ToListAsync();
+        }
+
+        // Method to get results with total votes greater than a specified value with related entities
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs]: remove message

File: src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs [hunk 19]
```diff
@@ -6,26 +6,67 @@
...
+        {
+            return await _dbSet
+                .Include(r => r.Elections)
+                .Include(r => r.Candidates)
+                .Where(r => r.TotalVotes > votes)
+                .ToListAsync();
+        }
+
+
+        // Method to get a specific result by ID (uses GetResultWithDetailsAsync internally)
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Infrastructure/Repositories/ResultRepository.cs]: remove message

**Review by ArianitSGashi — APPROVED** [review]

**Pull request merged**